server/src/test/java/org/elasticsearch/index/shard/ShardGetServiceTests.java

```diff
@@ -250,7 +252,6 @@ public void testGetFromTranslog() throws IOException {
         engine.flush(true, true, flushFuture);
         var flushResult = flushFuture.actionGet();
         assertTrue(flushResult.flushPerformed());
-        assertThat(flushResult.generation(), equalTo(lastUnsafeGeneration));
```

Comment:
Instead of removing this, would it work to instead assert that flush result (or current generation) is 1 higher than lastUnsafeGeneration? (since the flush is forced, even w/o a new indexing operation, we should get a new generation)